c10/core/impl/alloc_cpu.h

```diff
@@ -4,8 +4,21 @@
 
 #include <cstddef>
 
+#ifdef __linux__
+#include <sys/mman.h>
+#include <cstdlib>
+#endif
+
 namespace c10 {
 
+#ifdef __linux__
+// since the default thp pagesize is 2MB, enable thp only
+// for buffers of size 2MB or larger to avoid memory bloating
+constexpr size_t gAlloc_threshold_thp = 2 * 1024 * 1024;
```

Comment:
This constexpr seems to belong in alignment.h.
And why should it be guarded under `#ifdef __linux__` as transparent huge pages are hardware independent?

Reply:
we are not enabling this feature on other OSes, to avoid unused variable warnings, i'm defining them under __linux__ only